src/diffusers/loaders/ip_adapter.py

```diff
@@ -222,7 +222,7 @@ def load_ip_adapter(
 
             # create feature extractor if it has not been registered to the pipeline yet
             if hasattr(self, "feature_extractor") and getattr(self, "feature_extractor", None) is None:
-                clip_image_size = self.image_encoder.config.image_size
+                clip_image_size = self.image_encoder.config.image_size if self.image_encoder is not None else 224
```

Comment:
sure, I'll add the test, if we had it this wouldn't have passed without notice.

Reply:
changed my mind for at least this PR, we don't have a test for the image encoder or feature extractor, the tests just load the ip adapter weights and use `ip_adapter_image_embeds` . The whole `load_ip_adapter` function isn't covered in the tests.

I'll add in my TODO list to make a complete test for this later.

Reply:
Okay with that

Reply:
hi, thanks for adding me here, the change looks good to me

I just leave one comment: the `load_ip_adapter` function is covered in the slow tests in tests/pipelines/ip_adapters/test_ip_adapter_stable_diffusion.py. the same for the  `unload_ip_adapter` function. maybe it is better to test all of them in the fast tests (tests/pipelines/test_pipelines_common.py)?

Reply:
oh yeah missed that one, you're correct, the only caveat is that in the slow tests we don't test the FaceID one. 

> maybe it is better to test all of them in the fast tests

IMO yes, we can use tiny versions like the other tests and use those functions instead of just using the `ip_adapter_image_embeds`  so we can catch any loading errors in the fast tests too.

Reply:
oh sorry, but I see the but is on a different model, the Face ID Plus, that always requires the CLIP image encoder [docs - see the last code snippet](https://huggingface.co/docs/diffusers/main/en/using-diffusers/loading_adapters#ip-adapter-face-id-models).

Reply:
it is not a real bug, it is supposed to work in this way

Reply:
what do you mean? In my understanding this only happens with the FaceID ones and not the FaceID Plus ones, if there's a image encoder present, the code from before this fix runs ok.

Reply:
I see what you mean, the original issue was with a Plus Model, but still this fix is needed right when using the normal FaceID one right?

Reply:
> but still this fix is needed when using the normal FaceID one right

Sure

Reply:
> I see what you mean, the original issue was with a Plus Model, but still this fix is needed when using the normal FaceID one right?

exactly

Reply:
> Face ID Plus, that always requires the CLIP image encoder

Duplicating the answer on this: https://github.com/huggingface/diffusers/issues/9187#issuecomment-2293214533

So it's not actually strictly required and one can use the normal FaceID or FaceID Plus without `image_encoder`